tests/extmod/vfs_posix_ilistdir_del.py

```diff
@@ -0,0 +1,40 @@
+# Test ilistdir __del__ for VfsLittle using a RAM device
+import gc
+
+try:
+    import os
+
+    os.VfsPosix
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+
+def test(testdir):
+    vfs = os.VfsPosix(testdir)
+    vfs.mkdir("/test_d1")
+    vfs.mkdir("/test_d2")
+    vfs.mkdir("/test_d3")
+
+    for i in range(10):
+        print(i)
+        idir = vfs.ilistdir("/")
+        print(any(idir))
```

Comment:
This deserves a comment, eg: "we want to partially iterate the ilistdir iterator to leave it in an open state, which will then test the finaliser when it's GC'd"

Reply:
(also same comment in other 2 tests)